### PR TITLE
[DependencyScan] Fix and cleanup CAS Depscanner

### PR DESCRIPTION
**Pull request by cachemeifyoucan**
Fix and cleanup dependency scanner for CAS.

### CODE REVIEW

**Comment by cachemeifyoucan** [comment]
@artemcm Can you double check if my assumption in the last commit is correct? Currently, swift-driver will gets more and more command-line arguments as it tries to rescan the dependency.

File: include/swift/AST/ModuleDependencies.h [hunk 1]
```diff
@@ -138,6 +138,7 @@ class ModuleDependencyInfoStorageBase {
   std::string moduleCacheKey;
 
   bool resolved;
+  bool finalized;
```
Comment (artemcm) [inline include/swift/AST/ModuleDependencies.h:141]: It's worth adding comments here on what each one means. 😬

**Comment by artemcm** [comment]
> @artemcm Can you double check if my assumption in the last commit is correct? Currently, swift-driver will gets more and more command-line arguments as it tries to rescan the dependency.

Yeah, makes sense to me, good catch!

File: lib/ClangImporter/ClangImporter.cpp [hunk 2]
```diff
@@ -1166,8 +1166,10 @@ ClangImporter::create(ASTContext &ctx,
   auto fileMapping = getClangInvocationFileMapping(ctx);
   // Wrap Swift's FS to allow Clang to override the working directory
   llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> VFS =
-      llvm::vfs::RedirectingFileSystem::create(
```
Comment (benlangmuir) [inline lib/ClangImporter/ClangImporter.cpp:1169]: Is the comment above still true if there are no redirected files?

Comment (cachemeifyoucan) [reply lib/ClangImporter/ClangImporter.cpp:1169]: I didn't actually notice that comment. I guess it is to make sure that the clang instance here will definitely get a different FS so clang can call `setWorkingDir` on the FS without interfering with swift? Let me give this more thoughts.

**Comment by cachemeifyoucan** [comment]
@swift-ci please smoke test

**Comment by cachemeifyoucan** [comment]
@swift-ci please smoke test

**Review by benlangmuir — APPROVED** [review]

**Comment by cachemeifyoucan** [comment]
@swift-ci please smoke test

**Comment by cachemeifyoucan** [comment]
@swift-ci please smoke test

**Comment by cachemeifyoucan** [comment]
@swift-ci please smoke test

**Comment by cachemeifyoucan** [comment]
@swift-ci please smoke test

**Comment by cachemeifyoucan** [comment]
@swift-ci please clean smoke test

**Comment by cachemeifyoucan** [comment]
@swift-ci please smoke test

**Comment by cachemeifyoucan** [comment]
Drop out the part that finalize the command-line to avoid multiple argument append since that seems to break test cases nondeterministically.

It is very small fix now. Merging.

**Pull request merged**